specs/phase1/shard-fork-choice.md

```diff
@@ -151,10 +151,11 @@ def on_shard_block(store: Store, shard_store: ShardStore, signed_shard_block: Si
 
     # Check shard parent exists
     assert shard_block.shard_parent_root in shard_store.block_states
-    pre_shard_state = shard_store.block_states[shard_block.shard_parent_root]
+    shard_parent_state = shard_store.block_states[shard_block.shard_parent_root]
 
     # Check beacon parent exists
     assert shard_block.beacon_parent_root in store.block_states
+    beacon_parent_state = store.block_states[shard_block.beacon_parent_root]
```

Comment:
It's not valid when fail-crosslinking 

(added some missing lines)
![image](https://user-images.githubusercontent.com/9263930/83923087-19fa3000-a7b4-11ea-9522-2d0fc310dc27.png)


In this case, when we processing shard block `S_101`, the `beacon_parent_state` is at `B_101`, where `B_101.shard_states[shard_store.shard].slot` is 99 instead of 100.

Reply:
Right! my bad